lametro/search_indexes.py

```diff
@@ -50,8 +51,18 @@ def prepare_attachment_text(self, obj):
         )
 
     def prepare_legislative_session(self, obj):
-        start_year = obj.legislative_session.identifier
-        end_year = int(start_year) + 1
+        if len(obj.actions_and_agendas) > 0:
+            most_recent = sorted(obj.actions_and_agendas, key=lambda i: i['date'],reverse=True)[0]
+            action_date = most_recent['event'].start_time
```

Comment:
This isn't quite right. The desired logic is to return the session associated with the most recent agenda date. If there are no agenda dates, return the session associated with the most recent action date. If there is neither an agenda date or an action date, return None.

Reply:
Can you propose how you'd go about getting the desired date, using the output of `obj.actions_and_agendas`?

Reply:
As I'm thinking about it, I would have to add additional logic to check if the `most_recent` event is an agenda (`description` being `SCHEDULED`) or an action. I would use the agenda if possible, otherwise using an action as the `most_recent`.

Either way, I can use `most_recent['date']`, unless a `datetime` instance is preferable to a `date` instance for this?

Reply:
@hancush am I on the right track with this?

Reply:
I'd recommend something like:

```python
actions_and_agendas = sorted(obj.actions_and_agendas, key=lambda i: i['date'], reverse=True)
agendas = [item for item in actions_and_agendas if it is an agenda]  # will be sorted bc list order is preserved in python
if agendas:
    # return the most recent date from your agendas list
else:
    # there are no agendas, i.e., everything in actions_and_agendas is an action
    # return the most recent date from your actions_and_agendas list